apollo-router/src/axum_factory/axum_http_server_factory.rs

```diff
@@ -554,6 +567,14 @@ where
         router = router.route("/", get(handle_graphql::<RF>).post(handle_graphql::<RF>));
     }
 
+    // Tie the lifetime of the session count instrument to the lifetime of the router
+    // by moving it into a no-op layer.
+    let instrument = session_count_instrument();
+    router = router.layer(layer_fn(move |service| {
+        let _ = &instrument;
+        service
+    }));
```

Comment:
The `session_count_active` instrument is "stored" in an `axum::Router` instance by moving it into a no-op layer. Another option would be to just have one global instrument for the entire lifetime of the program, but I'm not sure if that would work everywhere we need it to (eg. does it do the right thing across reloads, tests, etc?)